core/src/main/scala/java/text/DecimalFormat.scala

```diff
@@ -1,57 +1,521 @@
+
+    // Round the target number based upon expected fractions, so we can compare it to the integer
+    val (targetNumber: JavaBigDecimal, expPower: Int) =
+      if (useScientificNotation) {
+        getExponentNumberAndPower(multiplied)
+      } else {
+        (multiplied.setScale(max(getMaximumFractionDigits(), getMinimumFractionDigits()), getRoundingMode), 0)
+      }
+
+    val integerStrBuilder = new StringBuilder()
+    val integerPart: JavaBigDecimal = new JavaBigDecimal(targetNumber.toBigInteger, 0)
+    var integerDigitsWritten: Int = 0
 
-  override final def format(obj: AnyRef, toAppendTo: StringBuffer, pos: FieldPosition): StringBuffer = ???
+    if ((integerPart.compareTo(JavaBigDecimal.ZERO) == 0) || (getMaximumIntegerDigits == 0)) {
+      integerStrBuilder.append(symbols.getZeroDigit)
+      integerDigitsWritten += 1
+    } else {
+      integerDigitsWritten += formatNumber(integerPart.toBigInteger, integerStrBuilder, true)
+    }
 
-  override final def parseObject(source: String, pos: ParsePosition): AnyRef = ???
+    // Add integer digit padding if needed
+    if (integerDigitsWritten < getMinimumIntegerDigits) {
+      (integerDigitsWritten until getMinimumIntegerDigits).foreach{ _ =>
+        handleGroupSeparator(integerStrBuilder, integerDigitsWritten)
+
+        integerStrBuilder.append(symbols.getZeroDigit)
+        integerDigitsWritten += 1
+      }
+    }
+
+    // We have the integer portion ready, append it to the builder
+    toAppendTo.append(integerStrBuilder.result.reverse)
+
+    // Compare the target number to the strictly-integer value...
+    targetNumber.compareTo(integerPart) match {
+      // We have a fractional value...
+      case 1  =>
+        toAppendTo.append(symbols.getDecimalSeparator)
+
+        val fractionStrBuilder = new StringBuilder()
+
+        // Special case for exponents
+        val fractionMaxDigits: Int = if (useScientificNotation) {
+          totalExponentPrecision - integerDigitsWritten
+        } else getMaximumFractionDigits
+
+        // Set scale to max fraction digits, subtract integer part, & get
+        // 12.0123 -> (set scale 5) 12.01230 -> (minus 12) 0.01230) -> unscaled value 1230
+        val fractionPart: JavaBigInteger =
+        targetNumber.setScale(fractionMaxDigits, getRoundingMode).subtract(integerPart).unscaledValue()
+
+        // Convert integer 1230 to a reversed string (0321)
+        formatNumber(fractionPart, fractionStrBuilder, false)
+
+        // 0321
+        val unscaledString = fractionStrBuilder.result
+
+        // Drop extra zero's at the end, then add significant '0's to end, then reverse it
+        // 0321 => 321 => 3210 => 0123
+        val fractionStr: String = {
+          val truncatedStr: String = unscaledString.dropWhile(_ == symbols.getZeroDigit)
+
+          truncatedStr +
+          (0 until (fractionMaxDigits - unscaledString.length)).map { _ =>
+            symbols.getZeroDigit
```

Comment:
This bit is probably wrong `getZeroDigit` referns to the pattern for 0 on the date format. It happens to be the char `0` so it mostly works. We should just use the value `0`, or even better the localized number `0`, However the latter is not available yet